bookwyrm/migrations/0171_hashtag_support.py

```diff
@@ -0,0 +1,53 @@
+# Generated by Django 3.2.16 on 2022-12-17 19:28
+
+import bookwyrm.models.fields
+import django.contrib.postgres.fields.citext
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("bookwyrm", "0170_merge_0168_auto_20221205_2331_0169_auto_20221206_0902"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="Hashtag",
+            fields=[
+                (
+                    "id",
+                    models.AutoField(
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                ("created_date", models.DateTimeField(auto_now_add=True)),
+                ("updated_date", models.DateTimeField(auto_now=True)),
+                (
+                    "remote_id",
```

Comment:
note that the remote id currently is just using the default, which renders this as https://your.domain.here/hashtag/1

This should probably be slugify the hashtag and use something like `/tags/MyHashtag`. I did some quick tests with Django's `slugify` and it seems that it isn't able to slugify at least Japanese characters (just returns an empty string). Maybe something like [unidecode](https://pypi.org/project/Unidecode/) would be better for this?

Reply:
Another option here would be to use decorative urls, like we do with author names and book titles - the domain looks like `https://your.domain.here/hashtag/1/MyHashtag`, but the slugified part is decorative, so if it doesn't render for whatever reason, the url still works. Relevant code:
https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/urls.py#L679
https://github.com/bookwyrm-social/bookwyrm/blob/d86e6a6949f720ce1ba2eb0ab440f289e2c19cdb/bookwyrm/views/author.py#L29

Reply:
ah that's a good solution, thank you!